pulsar_data_collection/models.py

```diff
@@ -0,0 +1,36 @@
+import datetime
+from typing import Any, Dict, List, Optional, Union
+
+import pandas as pd
+from config import factories
+from pydantic import BaseModel, validator
+
+
+class PulseParameters(BaseModel):
+    model_id: str
+    model_version: str
+    data_id: str
+    reference_data_storage: Any  # Union[str, Dict[str, str]]
+    target_name: str
+    storage_engine: str
+    login: Dict[str, Union[str, int, bool]]
+    features_metadata: Optional[Dict[str, type]]  # key: feature_name, value : feature type
+    other_labels: Optional[Dict[str, Union[str, int, bool]]] = None
+    timestamp_column_name: str = "timestamp"
+
+    @validator("storage_engine")
+    def check_storage_engine(cls, value):
```

Comment:
define the API with clarity
If your interest is to check 
def is_value_in_searchengine(cls, value)
searchstatus  = True  if value  in factories else False

if your interest is to get the value from searchengine
def get_value__from_searchengine(cls, value):
 if value not  in factories:
	raise ValueError(f"Storage Engine for {value} not supported")
return value